package.json

```diff
@@ -2,69 +2,67 @@
-    "postinstall": "node ./upgrade-warning"
-  },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/firebase/firebase-functions.git"
-  },
   "keywords": [
     "firebase",
     "functions",
     "google",
     "cloud"
   ],
-  "author": "Firebase Team",
-  "license": "MIT",
+  "homepage": "https://github.com/firebase/firebase-functions#readme",
   "bugs": {
     "url": "https://github.com/firebase/firebase-functions/issues"
   },
-  "homepage": "https://github.com/firebase/firebase-functions#readme",
-  "devDependencies": {
-    "@types/chai": "^3.4.32",
-    "@types/chai-as-promised": "0.0.28",
-    "@types/mocha": "^5.2.5",
-    "@types/mock-require": "^1.3.3",
-    "@types/nock": "^0.54.32",
-    "@types/node": "^6.0.38",
-    "@types/sinon": "^1.16.29",
-    "chai": "^3.5.0",
-    "chai-as-promised": "^5.2.0",
-    "firebase-admin": "^7.0.0",
-    "istanbul": "^0.4.2",
-    "mocha": "^5.2.0",
-    "mock-require": "^2.0.1",
-    "nock": "^9.0.0",
-    "prettier": "^1.13.7",
-    "sinon": "^1.17.4",
-    "ts-node": "^7.0.1",
-    "typescript": "~3.1.0"
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/firebase/firebase-functions.git"
   },
-  "peerDependencies": {
-    "firebase-admin": "^7.0.0"
+  "license": "MIT",
+  "author": "Firebase Team",
+  "main": "lib/index.js",
+  "types": "lib/index.d.ts",
+  "scripts": {
+    "build:pack": "rm -rf lib && npm install && tsc -p tsconfig.release.json && npm pack",
+    "build:release": "npm install --production && npm install typescript firebase-admin && tsc -p tsconfig.release.json",
+    "build": "tsc -p tsconfig.release.json",
+    "format": "prettier --write '**/*.ts'",
+    "posttest": "npm run format",
+    "test": "mocha -r ts-node/register ./spec/index.spec.ts"
   },
   "dependencies": {
-    "@types/cors": "^2.8.1",
-    "@types/express": "^4.11.1",
-    "@types/jsonwebtoken": "^7.2.6",
-    "@types/lodash": "^4.14.34",
     "cors": "^2.8.4",
-    "express": "^4.16.2",
-    "jsonwebtoken": "^8.2.1",
+    "express": "^4.17.1",
+    "jsonwebtoken": "^8.3.2",
     "lodash": "^4.6.1"
   },
-  "engines": {
-    "node": ">=6.0.0"
+  "devDependencies": {
+    "@types/chai": "^4.1.7",
+    "@types/chai-as-promised": "^7.1.0",
+    "@types/cors": "^2.8.5",
+    "@types/express": "^4.11.1",
+    "@types/jsonwebtoken": "^8.3.2",
+    "@types/lodash": "^4.14.132",
+    "@types/mocha": "^5.2.7",
+    "@types/mock-require": "^2.0.0",
+    "@types/nock": "^10.0.2",
+    "@types/node": "^8.10.49",
+    "@types/sinon": "^7.0.12",
+    "chai": "^4.2.0",
+    "chai-as-promised": "^7.1.1",
+    "firebase-admin": "^8.0.0",
+    "istanbul": "^0.4.2",
+    "mocha": "^6.1.4",
+    "mock-require": "^3.0.3",
+    "nock": "^10.0.6",
+    "prettier": "^1.17.1",
+    "sinon": "^7.3.2",
+    "ts-node": "^8.2.0",
+    "typescript": "^3.5.1"
   },
-  "typings": "lib/index.d.ts"
+  "peerDependencies": {
+    "firebase-admin": "^8.0.0",
+    "firebase-tools": "^5.0.0"
```

Comment:
Why was `firebase-tools` added here?

Reply:
My understanding is that firebase-functions is always used alongside firebase-tools. I thought it might be useful to specify which version of firebase-tools is supported.

As an example, this PR drops support for FIREBASE_PROJECT environment variable used in firebase-tools v3.18.

Reply:
Im torn here - I like the helpful warnings that tell the user they are using an outdated version of the Firebase CLI that is not compatible with this version of the functions SDK. I can see this reducing support requests. On the other hand, I can see some people using this without the Firebase CLI (like deploying with gcloud for example) and that may confuse them that the CLI is a requirement. However, deploying with gcloud is also not very well supported since most functions will require the FIREBASE_CONFIG env var which is only set if you deploy through Firebase CLI. @mbleigh what are your thoughts on this?

Reply:
One thing to note is that all `peerDependencies` are optional, so people using it with `gcloud` will not experience errors, just warnings during `npm install`. I'd guess a majority of developers uses it with `firebase-tools`?

Reply:
Discussed this with the team - let's leave out `firebase-tools` peer dependency since the functions SDK really does not depend on it like it does on `firebase-admin` and it may cause more trouble down the line.

Reply:
Done in 68018f8794965985f7ed1670b4cf32e91da46754